### PR TITLE
SceneryZ value identified and applied correctly

### PR DESCRIPTION
**Pull request by Chaosmeister**
Add the SceneryZ-value to the TrackZ value to compute the correct value for window_track_place_get_base_z().

Solves https://github.com/OpenRCT2/OpenRCT2/issues/259

### CODE REVIEW

File: src/openrct2/ride/track_design.c [hunk 1]
```diff
@@ -1312,7 +1312,7 @@ sint32 place_virtual_track(rct_track_td6 *td6, uint8 bl, uint8 rideIndex, sint32
     }
 
     if (bl == 3) {
-        return _trackDesignPlaceZ;
+		return _trackDesignPlaceZ -_trackDesignPlaceSceneryZ;
```
Comment (janisozaur) [inline src/openrct2/ride/track_design.c]: invalid indent

File: src/openrct2/ride/track_design.c [hunk 2]
```diff
@@ -1312,7 +1312,7 @@ sint32 place_virtual_track(rct_track_td6 *td6, uint8 bl, uint8 rideIndex, sint32
     }
 
     if (bl == 3) {
-        return _trackDesignPlaceZ;
+	 return _trackDesignPlaceZ -_trackDesignPlaceSceneryZ;
```
Comment (janisozaur) [inline src/openrct2/ride/track_design.c]: it should be spaces only, right now it's tab + space

File: src/openrct2/ride/track_design.c [hunk 3]
```diff
@@ -1312,7 +1312,7 @@ sint32 place_virtual_track(rct_track_td6 *td6, uint8 bl, uint8 rideIndex, sint32
     }
 
     if (bl == 3) {
-        return _trackDesignPlaceZ;
+         return _trackDesignPlaceZ - _trackDesignPlaceSceneryZ;
```
Comment (Gymnasiast) [inline src/openrct2/ride/track_design.c]: You accidentally put a space before this return statement.
Also, I'd like to see the following comment above this statement:
```
// Change from vanilla: originally, _trackDesignPlaceSceneryZ was not subtracted
// from _trackDesignPlaceZ, causing bug #259.
```

**Comment by Gymnasiast** [comment]
Tested it. It solves both the exploit in #259, as well as the crash that happened when taking the exploit too far. :+1:

**Pull request merged**